backend/src/zac/camunda/select_documents/serializers.py

```diff
@@ -0,0 +1,112 @@
+from zac.camunda.user_tasks import Context, register, usertask_context_serializer
+from zac.contrib.dowc.constants import DocFileTypes
+from zac.contrib.dowc.utils import get_dowc_url
+from zac.core.camunda import get_process_zaak_url
+from zac.core.services import get_documenten, get_zaak
+
+
+class DocumentSerializer(APIModelSerializer):
+    read_url = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Document
+        fields = (
+            "beschrijving",
+            "bestandsnaam",
+            "bestandsomvang",
+            "url",
+            "read_url",
+            "versie",
+        )
+
+    def get_read_url(self, obj) -> str:
+        """
+        Create the document read url to facilitate opening the document
+        in a MS WebDAV client.
+        """
+        return get_dowc_url(obj, purpose=DocFileTypes.read)
+
+
+@dataclass
+class DocumentSelectContext(Context):
+    documents: List[Document]
+
+
+@usertask_context_serializer
+class DocumentSelectContextSerializer(APIModelSerializer):
+    documents = DocumentSerializer(many=True)
+
+    class Meta:
+        model = DocumentSelectContext
+        fields = ("documents",)
+
+
+#
+# Write serializer
+#
+
+
+class DocumentSelectTaskSerializer(serializers.Serializer):
+    """
+    Serializes the selected documents for the task.
+
+    Requires ``task`` to be in serializer ``context``.
+    """
+
+    selected_documents = serializers.ListField(
+        child=serializers.URLField(),
+        label=_("Selecteer de relevante documenten"),
+        help_text=_(
+            "Dit zijn de documenten die bij de zaak horen. Selecteer de relevante "
+            "documenten."
+        ),
+    )
+
+    def validate_selected_documents(self, selected_documents):
+        # Make sure selected documents are unique
+        selected_documents = list(dict.fromkeys(selected_documents))
+
+        # Get zaak documents to verify valid document selection
+        task = self.context["task"]
+        process_instance = get_process_instance(task.process_instance_id)
+        self.zaak_url = get_process_zaak_url(process_instance)
+        zaak = get_zaak(zaak_url=self.zaak_url)
+        documenten, rest = get_documenten(zaak)
+        valid_docs = [doc.url for doc in documenten]
+
+        invalid_docs = [doc for doc in selected_documents if not doc in valid_docs]
+        if invalid_docs:
+            raise serializers.ValidationError(
+                _(
+                    "Selected documents: {invalid_docs} are invalid. Please choose one of the "
+                    "following documents: {valid_docs}."
+                ).format(invalid_docs=invalid_docs, valid_docs=valid_docs),
+                code="invalid_choice",
+            )
+
+        return selected_documents
+
+    def get_process_variables(self) -> Dict:
+        """
+        #TODO: ?
+        """
+        return {}
+
+    def on_task_submission(self) -> NoReturn:
```

Comment:
https://docs.python.org/3/library/typing.html#typing.NoReturn

```suggestion
    def on_task_submission(self) -> None:
```